FrontEnd/src/BookingPage/BookingDB.js

```diff
@@ -0,0 +1,108 @@
+import { database } from "../UserAuthentication/FirebaseConfig";
+import {
+  query,
+  where,
+  collection,
+  doc,
+  getDocs,
+  setDoc,
+  getDoc,
+  onSnapshot,
+} from "firebase/firestore";
+
+const DATABASE_FOLDER_NAME = "users";
+const requestAppointment = async (
+  userUID,
+  vendorUID,
+  appointmentData,
+  userData
+) => {
+  try {
+    const Status = "pending";
+    const customerUsername = userData.UserName;
+    const vendorAppointmentDataWithStatus = {
+      ...appointmentData,
+      userUID,
+      customerUsername,
+      Status,
+    };
+    const userAppointmentDataWithStatus = {
+      ...appointmentData,
+      vendorUID,
+      customerUsername,
+      Status,
+    };
+
+    const userDocRef = doc(database, DATABASE_FOLDER_NAME, userUID);
+    const userAppointmentCollection = collection(userDocRef, "Appointments");
+    const userAppointmentDocRef = doc(
+      userAppointmentCollection,
+      userAppointmentDataWithStatus.appointmentTitle
+    );
+    await setDoc(userAppointmentDocRef, userAppointmentDataWithStatus);
+
+    const vendorDocRef = doc(database, DATABASE_FOLDER_NAME, vendorUID);
+    const vendorAppointmentCollection = collection(
+      vendorDocRef,
+      "Appointments"
+    );
+    const vendorAppointmentDocRef = doc(
+      vendorAppointmentCollection,
+      vendorAppointmentDataWithStatus.appointmentTitle
+    );
+    await setDoc(vendorAppointmentDocRef, vendorAppointmentDataWithStatus);
+  } catch (error) {
+    throw new Error(`Error booking an appointment: ${error.message}`);
+  }
+};
+
+const vendorAppointmentNotification = (vendorId, callback) => {
+  try {
+    if (vendorId === undefined) {
+      throw new Error(`Invalid userId`);
```

Comment:
Should this be 
```suggestion
      throw new Error(`Invalid vendorID`);
```
?